.github/workflows/compile-and-test-latest.yml

```diff
@@ -0,0 +1,26 @@
+name: compile-and-test-latest
```

Comment:
Why do we need this action if we also run code coverage in evergreen?

Reply:
Hah good point. I copied it from Travis, but forgot we had it on evergreen as well. I removed the coverage bit.

In general, the action isn't necessary. It is helpful for external contributors to have some assurance their changes work, since they cannot trigger evergreen patch builds themselves.